File: ros_tomm_mind/src/realsense/cpp_src/camera_pub.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <image_transport/image_transport.h>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/opencv.hpp>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "usb_camera_pub_node");
  ros::NodeHandle nh;
  ROS_INFO("usb_camera_pub_node start");

  int video_device = 0;
  cv::VideoCapture cap(video_device, cv::CAP_V4L2);

  image_transport::ImageTransport it(nh);
  image_transport::Publisher image_pub = it.advertise("usb_camera/image_raw", 10);

  ros::Rate rate(100);

  while (ros::ok())
  {
    cv::Mat frame;
    cap >> frame;

    if (!frame.empty())
    {
      sensor_msgs::ImagePtr image_msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", frame).toImageMsg();
      image_pub.publish(image_msg);
    }

    ros::spinOnce();
    rate.sleep();
  }

  cap.release();
  return 0;
}

File: ros_tomm_mind/src/realsense/src/camera_pub.py
#! /usr/bin/env python

import cv2
from sensor_msgs.msg import Image
from cv_bridge import CvBridge
import rospy

def publish_usb_camera():
  rospy.init_node("usb_camera_pub_node", anonymous=True)
  rospy.loginfo("usb_camera_pub_node start")

  video_device = 0
  cap = cv2.VideoCapture(video_device, cv2.CAP_V4L2)

  image_pub = rospy.Publisher("usb_camera/image_raw", Image, queue_size=10)
  bridge = CvBridge()

  rate = rospy.Rate(100)

  while not rospy.is_shutdown():
    ret, frame = cap.read()

    if ret:
      image_msg = bridge.cv2_to_imgmsg(frame, encoding="bgr8")
      image_pub.publish(image_msg)

    rate.sleep()

  cap.release()


if __name__ == "__main__":
  try:
    publish_usb_camera()
  except rospy.ROSInterruptException:
    pass